react/src/component/login/email.tsx

```diff
@@ -70,7 +71,7 @@ export const LoginWithEmail = (props) => {
                         <Form.Control placeholder="email@example.com" value={email} onChange={(e) => updateEmail(e.target.value)}/>
                     </Col>
                     <Col sm="2">
-                        <Button variant="primary" onClick={() => doLogin(email)}>Continue</Button>
+                        <Button variant="primary" onClick={() => doLogin(email)} disabled={! (email && email.match(/^.+@.+\..+$/))}>Continue</Button>
```

Comment:
I would like to see us avoid code that looks like this. It's a rabbit hole of copied conditionals and broken regexes. I would think this logic could be conditional in Javascript, not JSX. There should be a function that returns whether an email address is valid and another function or value that says whether this button is disabled when an email is invalid. Indeed, any form OK button should disabled unless the fields are disabled. And here, the value is "Continue". It should be a standard and in a string table (with abstractions for sharing).